tensorflow/core/grappler/optimizers/auto_mixed_precision.cc

```diff
@@ -2291,6 +2291,13 @@ Status AutoMixedPrecision::Optimize(Cluster* cluster, const GrapplerItem& item,
     return OkStatus();
   }
 
+  if (num_gpus >= 1 && mode_ == AutoMixedPrecisionMode::BF16) {
```

Comment:
I don't think you should skip optimizing if there is a GPU. What if the user has a GPU attached but is still running their model on the CPU using oneDNN? I would remove the check, or just give a warning.

Reply:
Thanks for pointing that out. I have fixed it by just a warning log.